app/labor/cache_buster.rb

```diff
@@ -19,17 +22,55 @@ def self.bust(path)
     #
     # https://github.com/fastly/fastly-ruby#efficient-purging
     return unless Rails.env.production?
-    return if ApplicationConfig["FASTLY_API_KEY"].blank?
 
-    HTTParty.post("https://api.fastly.com/purge/https://#{ApplicationConfig['APP_DOMAIN']}#{path}",
-                  headers: { "Fastly-Key" => ApplicationConfig["FASTLY_API_KEY"] })
-    HTTParty.post("https://api.fastly.com/purge/https://#{ApplicationConfig['APP_DOMAIN']}#{path}?i=i",
-                  headers: { "Fastly-Key" => ApplicationConfig["FASTLY_API_KEY"] })
+    if fastly_enabled?
+      bust_fastly_cache(path)
+    elsif nginx_enabled?
+      bust_nginx_cache(path)
+    else
+      raise ConfigurationError, "You cannot bust a cache without a caching service! Please enable a caching service."
+    end
+  rescue ConfigurationError => e
+    Rails.logger.error(e)
   rescue URI::InvalidURIError => e
     Rails.logger.error("Trying to bust cache of an invalid uri: #{e}")
     DatadogStatsClient.increment("cache_buster.invalid_uri", tags: ["path:#{path}"])
   end
 
+  def self.fastly_enabled?
+    ApplicationConfig["FASTLY_API_KEY"].present? && ApplicationConfig["FASTLY_SERVICE_ID"].present?
+  end
+
+  def self.nginx_enabled?
+    ApplicationConfig["OPENRESTY_PROTOCOL"].present? && ApplicationConfig["OPENRESTY_DOMAIN"].present?
+  end
+
+  def self.bust_fastly_cache(path)
+    HTTParty.post(
+      "https://api.fastly.com/purge/https://#{ApplicationConfig['APP_DOMAIN']}#{path}",
+      headers: {
+        "Fastly-Key" => ApplicationConfig["FASTLY_API_KEY"]
+      },
+    )
+    HTTParty.post(
+      "https://api.fastly.com/purge/https://#{ApplicationConfig['APP_DOMAIN']}#{path}?i=i",
+      headers: {
+        "Fastly-Key" => ApplicationConfig["FASTLY_API_KEY"]
+      },
+    )
+  end
+
+  def self.bust_nginx_cache(path)
+    uri = URI.parse("#{ApplicationConfig['OPENRESTY_PROTOCOL']}#{ApplicationConfig['OPENRESTY_DOMAIN']}#{path}")
+    http = Net::HTTP.new(uri.host, uri.port)
+    response = http.request Net::HTTP::NginxPurge.new(uri.request_uri)
+
+    # Explicitly use `fail` here since we are not catching and re-raising the exception.
+    fail Error, response.body unless response.is_a?(Net::HTTPSuccess) # rubocop:disable Style/SignalException
```

Comment:
TIL about fail!
Does Honeybadger deal with this the same way it does raised exceptions?

Reply:
@mstruve I believe that `fail`, internally, [will `raise` a `RuntimeError`](https://apidock.com/ruby/Kernel/fail), so I would think that if Honeybadger _should_ be able to deal with it? (I couldn't find anything specific in the Honeybadger docs about any exceptions around `fail`).

To be 100% honest, I saw this usage [happening in `fastly-ruby`](https://github.com/fastly/fastly-ruby/blob/a2c5f2cf3247a07ee9c39fa7115ab8281e4b6239/lib/fastly/client.rb#L115), which is actually how _I_ learned about it! I hadn't heard of it or used it before either, and it seems like people are pretty divided on whether or not to use it. I kept it in because Fastly does that, and also because of [this](https://avdi.codes/jim-weirich-on-exceptions) blog post. But on the other hand, Rubocop does [_not_ seem to like or recommend this](https://github.com/rubocop-hq/ruby-style-guide/issues/512), so maybe we should just be `raise`ing here instead?

I'm down to change it if that's what people prefer, I kinda just threw it in here after seeing Fastly doing it 😛

Reply:
Oh, thanks for sharing those resources and that blog post! I think for the sake of code cohesion we should use raise here. I could see a younger dev coming in and seeing this and mimic it again with another fail and before you know it we have 50/50 fails and raises sprinkled throughout the codebase. Not a blocker but that is the way I would lean.

Reply:
> I could see a younger dev coming in and seeing this and mimic it again with another fail and before you know it we have 50/50 fails and raises sprinkled throughout the codebase.

Hm, I think that this is reason enough to convince me to use that instead 👍 Have switched this from `fail` to `raise` in the latest commit, which Honeybadger should catch for us should Nginx go down. Also threw in a nicer error message since we are `raise`ing the error ourselves now.